compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -408,6 +434,38 @@ def start(self):
     def _event_loop(self):
         self._command.start()
 
+        parent_identities = set()
+        if not is_privileged():
+            client_options = {
+                "funcx_service_address": self._config.funcx_service_address,
+                "environment": self._config.environment,
+            }
+            log.debug("Ascertaining user identity set (%s)", client_options)
+
+            gcc = GC.Client(**client_options)
+            try:
+                userinfo = gcc.login_manager.get_auth_client().userinfo()
+                ids = userinfo["identity_set"]
+                parent_identities.update(ident["sub"] for ident in ids)
+                log.debug(
+                    "User-endpoint start requests are valid from identites: %s",
+                    parent_identities,
+                )
+                del gcc, client_options, ids
+            except Exception as exc:
+                msg = "Failed to determine identity set; try `whoami` command?"
+                log.error(f"({type(exc).__name__}) {exc}\n    {msg}")
+                log.debug("Stopping; failed to determine identities", exc_info=exc)
+                self._time_to_stop = True
+                return
+
+            if not parent_identities:
+                # Not a privileged user -- we require a known identity to match
+                # start endpoint requests against.
+                log.error("Failed to determine identity set; try `whoami` command")
```

Comment:
Not really worth changing, but I always perk up at duplicate identical strings, especially user facing :)

Reply:
Hmm; I suppose I could check the `if not parent_identities` in the `try`, and raise there.  I'm going to take lunch, then will look at that.

Reply:
Have moved it into the try-branch.  I think that should satisfy your (good!) sense of propriety?